src/node/nodestate.h

```diff
@@ -1176,6 +1147,48 @@ namespace ccf
       open_frontend(ActorsType::users);
     }
 
+    void broadcast_ledger_secret(
+      Store::Tx& tx,
+      const LedgerSecret& secret,
+      kv::Version version = kv::NoVersion,
+      bool exclude_self = false)
+    {
+      GenesisGenerator g(network, tx);
+      auto secrets_view = tx.get_view(network.secrets);
+
+      auto trusted_nodes = g.get_trusted_nodes(
```

Comment:
I don't think `get_trusted_nodes()` should be a `GenesisGenerator` method, it has nothing to do with genesis and we don't call `broadcast_ledger_secret` only on node startup (afaik). I think it probably belongs in `nodestate` or somewhere else.

Reply:
Good point! As discussed, I will rename the `GenesisGenerator` in the next PR and make it a collection of static methods as it doesn't need to be stateful.